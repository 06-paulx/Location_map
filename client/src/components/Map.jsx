/* global window,google */
import React from 'react';

const Map = props => {
  window.initMap = () => {
    const map = new google.maps.Map(document.getElementById('map'), {
      center: { lat: props.lat, lng: props.lng },
      zoom: 16,
      scrollwheel: false,
    });

    const marker = new google.maps.Marker({
      position: { lat: props.lat, lng: props.lng },
      map: map,
      icon: '/css/map_icon.png',
    });
  };

  // const marker = google.maps.Marker({
  //   position: { lat: props.lat, lng: props.lng },
  //   map: map,
  //   icon: '/map_icon.png',
  // });

  // if (window.map && window.map.setCenter) {
  //   window.map.setCenter({ lat: props.lat, lng: props.lng });
  // }
  return <div className="map" id="map" />;
};
// <div className="map" id="map" />,

export default Map;
