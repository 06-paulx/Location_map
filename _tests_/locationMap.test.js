/* env jest */
/* global window */

import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Map from '../client/src/components/Map.jsx';

Enzyme.configure({ adapter: new Adapter() });

const request = require('supertest');
const app = require('../server');
const connection = require('../db/connection.js');

describe('GET /api/location/1', () => {
  test('It should respond with a listing object', done => {
    request(app)
      .get('/api/location/1')
      .end((err, response) => {
        expect(response.body.airport).toEqual(
          'San Francisco International Airport',
        );
        expect(response.body.city).toEqual('San Francisco, California');
        expect(response.body.listing_id).toEqual(1);
        expect(response.body.latitude).toBeGreaterThanOrEqual(37.706373 - 0.01);
        expect(response.body.latitude).toBeLessThanOrEqual(37.706373 + 0.01);
        expect(response.body.longitude).toBeLessThanOrEqual(-122.437089 + 0.01);
        expect(response.body.longitude).toBeGreaterThanOrEqual(
          -122.437089 - 0.01,
        );
        expect(response.statusCode).toBe(200);
        done();
      });
  });
  afterAll(() => {
    connection.end();
  });
});

describe('Map component rendering', () => {
  it('should render a map of the location', () => {
    const mapComponent = shallow(<Map />);
    expect(mapComponent).toMatchSnapshot();
  });
});

describe('Google initMap is called', () => {
  it('should call the Google initMap function', () => {
    const mapMock = jest.fn();
    const markerMock = jest.fn();
    window.google = { maps: { Map: mapMock, Marker: markerMock } };

    const lat = 37.784523;
    const lng = -122.356782;

    shallow(<Map lat={lat} lng={lng} />);
    window.initMap();
    expect(markerMock).toBeCalledWith({
      map: {},
      icon: '/css/map_icon.png',
      position: { lat: lat, lng: lng },
    });
    expect(mapMock).toBeCalledWith(window.document.getElementById('map'), {
      center: { lat: lat, lng: lng },
      zoom: 16,
      scrollwheel: false,
    });
  });
});
