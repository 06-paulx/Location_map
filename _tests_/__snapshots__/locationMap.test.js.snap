// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component rendering should render a map of the location 1`] = `
<div
  className="map"
  id="map"
/>
`;
